### PR TITLE
fix(typescript): do not generate environments file if no environments can be used

### PR DESCRIPTION
**Pull request by RohinBhargava**
## Changes

* Adds more robustness around environments config checks to see if the ts-generator should generate an environments file
* Modifies the `audiences` seed definition to accommodate new case, where a `partner` audience can be supplied downstream, that has no associated environment.
* Modifies the `ts-sdk` seed config to accommodate the new test definition structure to prove that the change is valid
* Bumps `Changelog` and `Version`
* Uncovered bugs in python and ts generators, with new `failing-audiences` test-definition and filed here: 
  * https://linear.app/buildwithfern/issue/FER-3045/[everyone]-duplicate-path-definitions-leads-to-bad-generation
  * https://linear.app/buildwithfern/issue/FER-3047/[seed]-failing-audiences-generates-the-wrong-type-in-ir

FER-2927

### CODE REVIEW

File: seed/ts-sdk/seed.yml [hunk 1]
```diff
@@ -31,6 +31,13 @@ fixtures:
         allowCustomFetcher: true
         timeoutInSeconds: 2
       outputFolder: allow-custom-fetcher
+  audiences:
```
Comment (RohinBhargava) [inline seed/ts-sdk/seed.yml:34]: relevant change here

File: test-definitions/fern/apis/audiences/definition/api.yml [hunk 2]
```diff
@@ -3,3 +3,12 @@ display-name: Audiences API
 audiences:
   - public
   - private
+environments:
```
Comment (RohinBhargava) [inline test-definitions/fern/apis/audiences/definition/api.yml:6]: and here

File: test-definitions/fern/apis/audiences/definition/api.yml [hunk 3]
```diff
@@ -3,3 +3,12 @@ display-name: Audiences API
...
+  environment-a:
```
Comment (RohinBhargava) [inline test-definitions/fern/apis/audiences/definition/api.yml:7]: and here

File: generators/typescript/sdk/environments-generator/src/EnvironmentsGenerator.ts [hunk 4]
```diff
@@ -20,7 +20,21 @@ export class EnvironmentsGenerator {
         environmentUrlsTypeName,
         environmentsConfig
     }: EnvironmentsGenerator.generateEnvironments.Args): GeneratedEnvironments {
-        if (environmentsConfig == null) {
+        if (
```
Comment (RohinBhargava) [inline generators/typescript/sdk/environments-generator/src/EnvironmentsGenerator.ts:23]: critical change

File: test-definitions/fern/apis/failing-audiences/definition/api.yml [hunk 5]
```diff
@@ -0,0 +1,16 @@
+name: audiences
```
Comment (dsinghvi) [inline test-definitions/fern/apis/failing-audiences/definition/api.yml:1]: It isn't easy to understand what edge case you are trying to test with `failing-audiences`. could there be a better name for this definition?

File: seed/ts-sdk/seed.yml [hunk 6]
```diff
@@ -31,6 +31,13 @@ fixtures:
         allowCustomFetcher: true
         timeoutInSeconds: 2
       outputFolder: allow-custom-fetcher
+  failing-audiences:
+    - customConfig: null
+      outputFolder: no-custom-config
+    - customConfig: null
+      audiences:
+        - partner
+      outputFolder: with-partner-audience
```
Comment (dsinghvi) [inline seed/ts-sdk/seed.yml:40]: is there a way to test these edge cases using the existing `audiences` fixture?

Comment (RohinBhargava) [reply seed/ts-sdk/seed.yml:40]: yeah! let me add both like this!

File: test-definitions/fern/apis/failing-audiences/definition/api.yml (hunk 5 above)
Comment (RohinBhargava) [reply test-definitions/fern/apis/failing-audiences/definition/api.yml:1]: This is just a temporary fixture, until the bugs get solved, and then it will be deleted!

**Review by dsinghvi — APPROVED** [review]
